src/rdf4cpp/rdf/query/Variable.hpp

```diff
@@ -15,6 +15,9 @@ class Variable : public Node {
     explicit Variable(std::string_view name, bool anonymous = false,
```

Comment:
While fixing some other things I came across this curious interface in `query::Variable`.
Isn't an anonymous variable supposed to _not_ have a name? But maybe I got that wrong.
I replicated this behaviour in the additional functions but found it weird.

Also, to me this ctor also seems like it could be removed in favor of the functions I added because 
```
auto v = query::Variable::make_anonymous("a"); // expresses intent
query::Variable v{"a", true}; // not very expressive, need to look at declaration of ctor
``` 
sidenote: while writing these two lines I originally got the meaning of the boolean wrong (i.e. true being named), even though I just looked at it a few minutes ago.

Reply:
Anonymous variables are strange design decision in SPARQL. They look like blank nodes but are actually variables that cannot be projected. As blank nodes, they start with `_:` followed by a blank node identifier. 

So the "name" is actually a blank node identifier